PyART/catalogs/icc.py

```diff
@@ -1,111 +1,209 @@
+        
+        # define self.psi4lm, self.t_psi4 and self.r_extr
+        self.load_psi4(tmax_after_peak=200)
+        
+        if integrate:
+            # get hlm and dhlm by psi4-integration
+            self.integr_opts = self.integrate_psi4(t_psi4=self.t_psi4, 
+                                                   radius=self.r_extr, 
+                                                   integr_opts=integr_opts, 
+                                                   M=1, modes=[(2,2)])
+            i0     = np.where(self.u>=0)[0][0] 
+            DeltaT = self.u[i0]-self.u[0]
+            self.cut(DeltaT, cut_dothlm=True, cut_psi4lm=False)
         else:
-            self.modes = self.ell_emms # TODO: add check on input
-        data = []
-        for i, meta in enumerate(self.catalog_meta):
-            spin_asum = abs(float(meta['initial-bh-spin1z'])) + abs(float(meta['initial-bh-spin2z']))
-            if not self.nonspinning or spin_asum<1e-14:
-                wave = WaveIntegrated(path=meta['path'], r_extr=meta['r0'], modes=self.modes, M=meta['M'],
-                                      integr_opts=integr_opts)
-                if load_puncts:
-                    tracks = self.load_tracks(path=meta['path'])
-                    punct0 = np.column_stack( (tracks['t'], tracks['t'], tracks['x0'], tracks['y0'], tracks['z0']) )
-                    punct1 = np.column_stack( (tracks['t'], tracks['t'], tracks['x1'], tracks['y1'], tracks['z1']) )
-                    scat_NR = ScatteringAngle(punct0=punct0, punct1=punct1, file_format='GRA', nmin=2, nmax=5, n_extract=4,
-                                           r_cutoff_out_low=25, r_cutoff_out_high=None,
-                                           r_cutoff_in_low=25, r_cutoff_in_high=100,
-                                           verbose=False)
-                    scat_info = {'chi':scat_NR.chi, 'chi_fit_err':scat_NR.fit_err}
-                else:
-                    tracks    = {}
-                    scat_info = {}
-                
-                sim_data            = lambda:0
-                sim_data.meta       = meta
-                sim_data.wave       = wave
-                sim_data.tracks     = tracks
-                sim_data.scat_info  = scat_info
-                data.append(sim_data)
-        self.data = data
+            self.integr_opts = {}
+            self.load_hlm_compute_dothlm()
+        
         pass
-    
-    def load_tracks(self, path):
-        # FIXME: very specific
-        fname = 'puncturetracker-pt_loc..asc'
-        X  = np.loadtxt(os.path.join(path,fname))
-        t  = X[:, 8]
-        x0 = X[:,22]
-        y0 = X[:,32]
-        x1 = X[:,23]
-        y1 = X[:,33]
-        x  = x0-x1
-        y  = y0-y1
-        r  = np.sqrt(x**2+y**2)
-        th = -np.unwrap(np.angle(x+1j*y))
-        return {'t':t, 'x0':x0, 'x1':x1, 'y0':y0, 'y1':y1, 'r':r, 'th':th, 'z0':0*t, 'z1':0*t}
 
-    def get_simlist(self):
+    def load_meta(self):
```

Comment:
For consistency with other waveform classes, this should be named `load_metadata` I think